### ISSUE TITLE
Timer exceeded memory limit with Segmentation fault on HHVM and Travis CI

### ISSUE THREAD

**Issue by photodude** [sebastianbergmann/phpunit#1714]
We have ran into the [following issue](https://travis-ci.org/joomla/joomla-cms/jobs/62108921) running PHPUnit 4.1.6 or phpunit 4.6.6 on Travis CI with HHVM (This might be running against phpunit 4.5 on Travis CI)
```
Fatal error: request has exceeded memory limit in /vendor/phpunit/php-timer/PHP/Timer.php on line 138
/home/travis/build.sh: line 41:  5463 Segmentation fault      phpunit --configuration travisci-phpunit.xml
```

**Comment by nik045** [comment]
:+1:

**Comment by sebastianbergmann** [comment]
* `request has exceeded memory limit` tells you that the execution of your test suite requires more memory than you allow in your HHVM configuration
* HHVM must not segfault, though. That, however, is an issue that should be reported on the HHVM project.

**Comment by photodude** [comment]
@sebastianbergmann I'm coming back to this issue

I''m now running PHPunit 4.8.28 and HHVM 3.13.1 I'm still [running into this issue](https://travis-ci.org/photodude/joomla-cms/jobs/127249333)
`Fatal error: request has exceeded memory limit in /home/travis/build/photodude/joomla-cms/libraries/vendor/phpunit/php-timer/src/Timer.php on line 97`

hhvm reported memory limit using ini_get("memory_limit") returns 17179869184 available (16gb?)

Which I would assume should be more than enough. Is there something else causing a problem?

**Comment by photodude** [comment]
Ok I figured out how to change the hhvm memory limit in travis-ci
in the .travis.yml
```yml
 - if [[ $TRAVIS_PHP_VERSION = hhvm ]]; then cat build/travis/phpenv/hhvm.php.ini >> /etc/hhvm/php.ini; fi
```

with the hhvm.php.ini set to unlimited memory (i.e. system max)
```ini
memory_limit = -1
```
so now the memory limit is listed as 9223372036854775807 (should be more that enough for everything)

but I'm still getting the out of memory error
`Fatal error: request has exceeded memory limit in /home/travis/build/photodude/joomla-cms/libraries/vendor/phpunit/php-timer/src/Timer.php on line 97`

Peak memory is still listed as 2097152
https://travis-ci.org/photodude/joomla-cms/jobs/127696426

**Comment by photodude** [comment]
Looks like the single test causing the out of memory error is ['FinderIndexerStemmerPorter_EnTest::testStemWrongLanguage'"](https://github.com/joomla/joomla-cms/blob/staging/tests/unit/suites/finderIndexer/stemmer/FinderIndexerStemmerPorter_EnTest.php)

**Comment by photodude** [comment]
Turns out this issue was due to a bug in HHVM and will be fixed in the HHVM 3.15.2 release.
